llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -161,13 +161,20 @@ static cl::opt<int>
 
 int AIEHazardRecognizer::NumInstrsScheduled = 0;
 
-AIEHazardRecognizer::AIEHazardRecognizer(const AIEBaseInstrInfo *TII,
-                                         const InstrItineraryData *II,
-                                         bool IsPreRA)
+AIEHazardRecognizer::AIEHazardRecognizer(
+    const AIEBaseInstrInfo *TII, const InstrItineraryData *II, bool IsPreRA,
+    std::optional<unsigned> ScoreboardDepth)
     : TII(TII), ItinData(II) {
 
-  computeMaxLatency();
-  int Depth = computeScoreboardDepth();
+  int Depth = 0;
+  if (ScoreboardDepth.has_value()) {
```

Comment:
It would be nice to have a short comment here, otherwise one can expect it to be a real use-case for our target.